subgraphs/aave-forks/src/mapping.ts

```diff
@@ -148,118 +222,157 @@ export function _handleCollateralConfigurationChanged(
+  storePrePauseState(market);
 }
 
 export function _handleReserveActivated(
-  marketId: Address,
+  asset: Address,
   protocolData: ProtocolData
 ): void {
-  const market = getOrCreateMarket(marketId, protocolData);
+  const market = getMarketFromToken(asset, protocolData);
+  if (!market) {
+    log.warning("[_handleReserveActivated] Market for token {} not found", [
+      asset.toHexString(),
+    ]);
+    return;
+  }
 
   market.isActive = true;
-  market._prePauseState = [
-    true,
-    market._prePauseState[1],
-    market._prePauseState[2],
-  ];
   market.save();
+  storePrePauseState(market);
 }
 
 export function _handleReserveDeactivated(
-  marketId: Address,
-  protocolData: ProtocolData
+  asset: Address,
+  procotolData: ProtocolData
 ): void {
-  const market = getOrCreateMarket(marketId, protocolData);
+  const market = getMarketFromToken(asset, procotolData);
+  if (!market) {
+    log.warning("[_handleReserveDeactivated] Market for token {} not found", [
+      asset.toHexString(),
+    ]);
+    return;
+  }
 
   market.isActive = false;
-  market._prePauseState = [
-    false,
-    market._prePauseState[1],
-    market._prePauseState[2],
-  ];
   market.save();
+  storePrePauseState(market);
 }
 
 export function _handleReserveFactorChanged(
-  marketId: Address,
+  asset: Address,
   reserveFactor: BigInt,
-  protocolData: ProtocolData
+  procotolData: ProtocolData
 ): void {
-  const market = getOrCreateMarket(marketId, protocolData);
+  const market = getMarketFromToken(asset, procotolData);
+  if (!market) {
+    log.warning("[_handleReserveFactorChanged] Market for token {} not found", [
+      asset.toHexString(),
+    ]);
+    return;
+  }
 
-  market._reserveFactor = reserveFactor
+  market.reserveFactor = reserveFactor
     .toBigDecimal()
-    .div(exponentToBigDecimal(INT_TWO));
+    .div(exponentToBigDecimal(INT_FOUR));
   market.save();
 }
 
-export function _handleReserveUsedAsCollateralEnabled(
-  marketId: Address,
-  accountID: Address,
-  protocolData: ProtocolData
+export function _handleLiquidationProtocolFeeChanged(
+  asset: Address,
+  liquidationProtocolFee: BigInt,
+  procotolData: ProtocolData
 ): void {
-  const protocol = getOrCreateLendingProtocol(protocolData);
-
-  const market = getOrCreateMarket(marketId, protocolData);
+  const market = getMarketFromToken(asset, procotolData);
+  if (!market) {
+    log.warning(
+      "[_handleLiquidationProtocolFeeChanged] Market for token {} not found",
+      [asset.toHexString()]
+    );
+    return;
+  }
 
-  // grab account
-  let account = Account.load(accountID.toHexString());
-  if (!account) {
-    account = createAccount(accountID.toHexString());
-    account.save();
+  market._liquidationProtocolFee = liquidationProtocolFee
+    .toBigDecimal()
+    .div(exponentToBigDecimal(INT_FOUR));
+  market.save();
+}
 
-    protocol.cumulativeUniqueUsers += 1;
-    protocol.save();
+export function _handleReserveUsedAsCollateralEnabled(
```

Comment:
These functions need to also set the `position.isCollateral` field

Reply:
I believe isCollateral is handled here: https://github.com/tnkrxyz/subgraphs/blob/0b84b4042ac0a8c9f16c38502f26166fce4ece89/subgraphs/aave-forks/src/mapping.ts#L694-L703

Reply:
See this [query](https://api.thegraph.com/subgraphs/name/tnkrxyz/aave-v3-ethereum/graphql?query=query+MyQuery+%7B%0A++positions%28where%3A+%7BisCollateral%3A+true%7D%29+%7B%0A++++id%0A++++side%0A++++type%0A++++balance%0A++++isCollateral%0A++%7D%0A%7D)

Reply:
there is a bug in checking _enabledCollaterals array. fixed in 
104d24d

[query deployment with fix](https://api.thegraph.com/subgraphs/id/QmdVZ8gguWt6kcCijoYGW9suF59JXsTdAQvDR37cua7N42/graphql?query=query+MyQuery+%7B%0A++positions%28where%3A+%7BisCollateral%3A+true%7D%29+%7B%0A++++id%0A++++side%0A++++type%0A++++balance%0A++++isCollateral%0A++%7D%0A%7D)